cmd/cli/uninstall_mesh.go

```diff
@@ -105,92 +107,140 @@ func newUninstallMeshCmd(config *action.Configuration, in io.Reader, out io.Writ
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
+	meshesToUninstall := []meshInfo{}
 
 	if !settings.IsManaged() {
-		if !d.force {
-			// print a list of meshes within the cluster for a better user experience
-			fmt.Fprintf(d.out, "\nList of meshes present in the cluster:\n")
-
-			listCmd := &meshListCmd{
-				out:       d.out,
-				config:    d.config,
-				clientSet: d.clientSet,
-				localPort: d.localPort,
+		meshInfoList, err := getMeshInfoList(d.config, d.clientSet)
+		if err != nil {
+			return errors.Wrapf(err, "unable to list meshes within the cluster")
+		}
+		if len(meshInfoList) == 0 {
+			fmt.Fprintf(d.out, "No OSM control planes found\n")
+			return nil
+		}
+		specifiedMeshFound := false
+		// Searches for the mesh specified by the mesh-name flag if specified
+		if d.meshName != defaultMeshName {
+			specifiedMeshFound = d.findMesh(meshInfoList)
+			if !specifiedMeshFound {
+				return nil
 			}
+		}
 
-			err := listCmd.run()
-
-			// Unable to list meshes in the cluster
+		// Adds the mesh to be force uninstalled
+		if d.force {
+			// For force uninstall, if single mesh in cluster, set default to that mesh
+			if len(meshInfoList) == 1 {
+				d.meshName = meshInfoList[0].name
+				d.meshNamespace = meshInfoList[0].namespace
+			}
+			forceMesh := meshInfo{name: d.meshName, namespace: d.meshNamespace}
+			meshesToUninstall = append(meshesToUninstall, forceMesh)
+		} else {
+			// print a list of meshes within the cluster for a better user experience
+			err := d.printMeshes()
 			if err != nil {
 				return err
 			}
+			// Prompts user on whether to uninstall each OSM mesh in the cluster
+			for _, mesh := range meshInfoList {
+				// Only prompt for specified mesh if `mesh-name` is specified
+				if specifiedMeshFound && mesh.name != d.meshName {
+					continue
+				}
+				confirm, err := confirm(d.in, d.out, fmt.Sprintf("\nUninstall OSM [mesh name: %s] in namespace [%s] and/or OSM resources?", mesh.name, mesh.namespace), 3)
+				if err != nil {
+					return err
+				}
+				if confirm {
+					meshesToUninstall = append(meshesToUninstall, mesh)
+				}
+			}
+		}
 
-			confirm, err := confirm(d.in, d.out, fmt.Sprintf("\nUninstall OSM [mesh name: %s] in namespace [%s] and/or OSM resources ?", d.meshName, d.meshNamespace), 3)
-			if !confirm || err != nil {
+		for _, m := range meshesToUninstall {
+			// Re-initializes uninstall config with the namespace of the mesh to be uninstalled
+			err := d.actionConfig.Init(settings.RESTClientGetter(), m.namespace, "secret", debug)
+			if err != nil {
 				return err
 			}
-		}
 
-		_, err := d.client.Run(d.meshName)
-		if err != nil && errors.Cause(err) == helmStorage.ErrReleaseNotFound {
-			fmt.Fprintf(d.out, "No OSM control plane with mesh name [%s] found in namespace [%s]\n", d.meshName, d.meshNamespace)
+			_, err = d.client.Run(m.name)
+			if err != nil {
+				if errors.Cause(err) == helmStorage.ErrReleaseNotFound {
```

Comment:
It looks like this is still silently passing over errors other than "not found" when the `--delete-cluster-wide-resources` or `--delete-namespace` flags are passed.